crates/sui-framework/packages/sui-framework/sources/crypto/zklogin.move

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+module sui::zklogin {
+    use std::option;
+    use std::option::Option;
+    use std::string::String;
+    use sui::tx_context::TxContext;
+    use sui::tx_context;
+
+    /// Error if any of the inputs are longer than the allowed upper bounds
+    const EInvalidInput: u64 = 0;
+
+    /// Posession of a VerifiedID proves that the uses's address was created using zklogin and the
+    /// given parameters
+    struct VerifiedID {
+        /// The name of the claim
+        kc_name: String,
+
+        /// The value of the claim
+        kc_value: String,
+
+        /// The issuer
+        iss: String,
+
+        /// The audience (wallet)
+        aud: String,
+    }
```

Comment:
Shall we add the address here as well? Might be redundant because this will also be in the ownership info, but I could imagine cases where it's convenient to read this from Move (which can't see the ownership info).